### DISCUSSION TITLE
Communication between stores in the same tree

### DISCUSSION THREAD

**Discussion by juanmavillarrazadb** [mobxjs/mobx-state-tree#1902] [Q&A]
Hi every body, this is my first time with a discussion here and I'm an argentinian guy xD, my apologize for the spelling mistakes, so if you don't understand something or whatever just let me know...

With my work team (a great one) are building a defi trading multi-chain platform (awesome project) and we decided to choose MST for our application state within react and many others techs.

We push to standardize and design our state of the application based on the business logic, so imagine the following scenario (little approach of what we're looking for in many parts of the app).

The app has a root store, that contains 2 different store, one for the `user` state, and the other for the `assets` state. The `user` one only has simple data and an `assetId` that will be used to "populate" some sort of market information from the `assets` one, like for example the market price through a method called `getById` defined on it.

Our idea is to consume this information from the `assets` store inside the `user` one with [Computed value](https://mobx-state-tree.js.org/concepts/views), to achieve this we try three different ways to access a method from an other store, with [getRoot](https://mobx-state-tree.js.org//API/#getroot), [getParent](https://mobx-state-tree.js.org//API/#getparent) or storing a [safereference](https://mobx-state-tree.js.org//API/#safereference). And here is the main problem, we are not able to access the last node state of the `assets` store inside the `user` one, even when the assets mocked data has been loaded. So we don't know if doing this is a good practice, something possible with this tech or there is something I'm not seeing... We find a "solution" which I believe that is ninja fix and I don't feel comfortable using it, executing a `getSnapshot(assets)` inside the `getter` and forcing MST to get the last state of the node giving us the expected behavior.

I also try some others alternatives that you can check on the example repo. Here is the link of a simple code base that reproduce our actual problem in the app: https://github.com/juanmavillarraza/mst-issue

**Comment by EmilTholin** [comment]
Hi @juanmavillarrazadb!

What is the end goal you are trying to achieve with `getById` on the `UserStore`? Maybe that logic could be just on the `AssetsStore` instead?

**Reply by juanmavillarrazadb** [reply]
Hi @EmilTholin 

As you said `getById` is a method from the `AssetsStore` check this line https://github.com/juanmavillarraza/mst-issue/blob/main/src/stores/assets.ts#L46

**Reply by juanmavillarrazadb** [reply]
Let me know if you have any doubt, also if you can reproduce the error in this lines https://github.com/juanmavillarraza/mst-issue/blob/main/src/components/pages/Home/Home.tsx#L12
